widget/list.go

```diff
@@ -0,0 +1,405 @@
+		l.children = l.children[:len(l.children)-1]
+		l.firstItemIndex--
+		l.lastItemIndex--
+		l.prependItem(l.firstItemIndex)
+	}
+}
+
+func (l *listRenderer) setupListItem(item fyne.CanvasObject, index int) {
+	if index != l.list.selectedIndex {
+		item.(*listItem).selected = false
+	} else {
+		item.(*listItem).selected = true
+		l.list.selectedItem = item.(*listItem)
+	}
+	l.list.UpdateItem(index, item.(*listItem).child)
+	item.(*listItem).onTapped = func() {
+		if l.list.selectedItem != nil && l.list.selectedIndex >= l.firstItemIndex && l.list.selectedIndex <= l.lastItemIndex {
+			l.list.selectedItem.selected = false
+			l.list.selectedItem.Refresh()
+		}
+		l.list.selectedItem = item.(*listItem)
+		l.list.selectedIndex = index
+		l.list.selectedItem.selected = true
+		l.list.selectedItem.Refresh()
+		if l.list.OnItemSelected != nil {
+			l.list.OnItemSelected(index, item.(*listItem).child)
+		}
+	}
+	item.Refresh()
+}
+
+type listItem struct {
+	DisableableWidget
+
+	onTapped          func()
+	statusIndicator   *canvas.Rectangle
+	child             fyne.CanvasObject
+	hovered, selected bool
+}
+
+func newListItem(child fyne.CanvasObject, tapped func()) *listItem {
+	li := &listItem{
+		child:    child,
+		onTapped: tapped,
+	}
+
+	li.ExtendBaseWidget(li)
+	return li
+}
+
+// CreateRenderer is a private method to Fyne which links this widget to its renderer
+func (li *listItem) CreateRenderer() fyne.WidgetRenderer {
+	li.ExtendBaseWidget(li)
+
+	li.statusIndicator = canvas.NewRectangle(theme.BackgroundColor())
+
+	objects := []fyne.CanvasObject{li.statusIndicator, li.child}
+
+	return &listItemRenderer{widget.NewShadowingRenderer(objects, widget.ButtonLevel), li, layout.NewHBoxLayout()}
+}
+
+// MinSize returns the size that this widget should not shrink below
+func (li *listItem) MinSize() fyne.Size {
+	li.ExtendBaseWidget(li)
+	return li.BaseWidget.MinSize()
+}
+
+// MouseIn is called when a desktop pointer enters the widget
+func (li *listItem) MouseIn(*desktop.MouseEvent) {
+	li.hovered = true
+	li.Refresh()
+}
+
+// MouseMoved is called when a desktop pointer hovers over the widget
+func (li *listItem) MouseMoved(*desktop.MouseEvent) {
+}
+
+// MouseOut is called when a desktop pointer exits the widget
+func (li *listItem) MouseOut() {
+	li.hovered = false
+	li.Refresh()
+}
+
+// Tapped is called when a pointer tapped event is captured and triggers any tap handler
+func (li *listItem) Tapped(*fyne.PointEvent) {
+	if li.onTapped != nil && !li.Disabled() {
+		li.selected = true
+		li.Refresh()
+		li.onTapped()
+	}
+}
+
+type listItemRenderer struct {
+	*widget.ShadowingRenderer
+
+	item   *listItem
+	layout fyne.Layout
+}
+
+// MinSize calculates the minimum size of a listItem.
+// This is based on the size of the status indicator and the size of the child object.
+func (li *listItemRenderer) MinSize() (size fyne.Size) {
+	li.item.statusIndicator.SetMinSize(fyne.NewSize(theme.Padding(), li.item.child.Size().Height))
```

Comment:
setting minsize in the call to MinSize() seems strange

Reply:
This has been Resolved.